src/lib/Export.hs

```diff
@@ -139,21 +137,21 @@ prepareFunctionForExport f = do
       where
         go shape = \case
           BaseTy (Scalar sbt) -> Just $ RectContArrayPtr sbt shape
-          TabTy  (b:>Fin n) a -> do
-            dim <- case n of
-              Var v       -> Just (Left v)
-              IdxRepVal s -> Just (Right $ fromIntegral s)
+          TabTy  (b:>ixTy) a -> do
+            case ixTypeType ixTy of
+              Var _       -> return ()
+              IdxRepVal _ -> return ()
               _           -> Nothing
             case hoist b a of
-              HoistSuccess a' -> go (shape ++ [dim]) a'
+              HoistSuccess a' -> go (shape ++ [ixTy]) a'
               HoistFailure _  -> Nothing
           _ -> Nothing
 {-# SCC prepareFunctionForExport #-}
 
 -- === Exported function signature ===
 
 data ArgVisibility = ImplicitArg | ExplicitArg
-data ExportType n = RectContArrayPtr ScalarBaseType [Either (AtomName n) Int]
+data ExportType n = RectContArrayPtr ScalarBaseType [IxType n]
```

Comment:
I'd really like this representation to correspond to the on the wire format, which doesn't support arbitrary `ixType`s. Please use the old one

Reply:
Sounds good. I wasn't sure if it was just something that hadn't been implemented yet. Implementing the general case was easier. But I can restrict it